File: src/platform2/libsegmentation/feature_check.cc
// Copyright 2023 The ChromiumOS Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
//
// simple executable to encapsulation libsegmentation library to check from the
// command line if a feature is enabled. The commands are purposely limited as
// this executable is installed on all images.

#include <iostream>

#include <brillo/flag_helper.h>
#include <base/logging.h>
#include <libsegmentation/feature_management.h>

int main(int argc, char* argv[]) {
  DEFINE_string(feature_name, "", "return true when the feature is supported");
  brillo::FlagHelper::Init(argc, argv, "Query the segmentation library");

  if (FLAGS_feature_name.empty()) {
    LOG(ERROR) << "feature_name is a required argument";
    return 1;
  }

  segmentation::FeatureManagement feature_management;
  std::cout << feature_management.IsFeatureEnabled(FLAGS_feature_name)
            << std::endl;

  return 0;
}

File: src/platform2/featured/rust-client/examples/feature_check.rs
// Copyright 2022 The ChromiumOS Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use featured::{CheckFeature, Feature, PlatformFeatures};
use log::info;

fn main() {
    let features = PlatformFeatures::get().expect("Unable to get client");

    let feature =
        Feature::new("CrOSLateBootMyAwesomeFeature", false).expect("Unable to create feature");
    let is_enabled = features.is_feature_enabled_blocking(&feature);
    info!("Feature is enabled: {}", is_enabled);

    let status = features
        .get_params_and_enabled(&[&feature])
        .expect("Unable to fetch feature status");
    info!("Feature params: {:?}", status.get_params(&feature).unwrap());
}
